qt/widgets/common/src/PropertyWidget.cpp

```diff
@@ -40,15 +38,8 @@ double stringToRoundedNumber(const std::string &s) {
   // warning).
   const bool containsComma = s.find(",") != std::string::npos;
   if (containsComma)
-    throw std::runtime_error("");
-
-  std::istringstream i(s);
-  double roundedNumber;
-
-  if (!(i >> roundedNumber))
-    throw std::runtime_error("");
-
-  return roundedNumber;
+    throw std::runtime_error("string contains a comma");
+  return std::stod(s);
```

Comment:
I suspect `std::stod` is throwing a `std::out_of_range` exception, which doesn't inherit from `std::runtime_error` and could be causing the doctest failure.

Reply:
You're right. Applying the following diff

```diff
diff --git a/qt/python/mantidqtpython/mantidqt.sip b/qt/python/mantidqtpython/mantidqt.sip
index 249af60..d119fdd 100644
--- a/qt/python/mantidqtpython/mantidqt.sip
+++ b/qt/python/mantidqtpython/mantidqt.sip
@@ -162,7 +162,7 @@ class InterfaceManager
 %End
 public:
   MantidQt::API::AlgorithmDialog* createDialogFromName(const QString&, const int = -1,
-                                                       QWidget* = 0, bool = false);
+                                                       QWidget* = 0, bool = false) throw(std::exception);
   MantidQt::API::UserSubWindow* createSubWindow(const QString& interface_name, QWidget* parent = 0);
 
   void showHelpPage(const QString & url=QString());
```
gives a better (still not great) error message in Python. It at least matches the C++ one.